lib/modules/manager/maven/update.ts

```diff
@@ -78,7 +79,24 @@ export function bumpPackageVersion(
     const startTagPosition = versionNode.startTagPosition;
     const versionPosition = content.indexOf(versionNode.val, startTagPosition);
 
-    const newPomVersion = semver.inc(currentValue, bumpVersion);
+    let newPomVersion: string | null = null;
+    const prerelease = semver.prerelease(currentValue);
+    if (!prerelease) {
```

Comment:
```suggestion
    if (isSnapshot(currentPreleaseValue)) {
       // ...
    } else if (!currentPreleaseValue) {
```

Reply:
The problem we have right now is that the code is hard to understand, which will mean even harder to maintain in future. Putting the "true" part first is generally best practice (i.e. it's bad practice to say "if (!foo") {} else {}"